source/source_test.go

```diff
@@ -0,0 +1,186 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package source
+
+import (
+	"context"
+	"errors"
+	"net/url"
+	"testing"
+	"time"
+
+	"github.com/conduitio-labs/conduit-connector-mongo/config"
+	"github.com/conduitio-labs/conduit-connector-mongo/source/mock"
+	sdk "github.com/conduitio/conduit-connector-sdk"
+	"github.com/golang/mock/gomock"
+	"github.com/matryer/is"
+)
+
+func TestSource_Configure_success(t *testing.T) {
+	t.Parallel()
+
+	is := is.New(t)
+
+	s := Source{}
+
+	err := s.Configure(context.Background(), map[string]string{
+		config.KeyURI:        "mongodb://localhost:27017",
+		config.KeyDB:         "test",
+		config.KeyCollection: "users",
+	})
+	is.NoErr(err)
+	is.Equal(s.config, Config{
+		Config: config.Config{
+			URI: &url.URL{
+				Scheme: "mongodb",
+				Host:   "localhost:27017",
+			},
+			DB:         "test",
+			Collection: "users",
+		},
+		BatchSize:     defaultBatchSize,
+		Snapshot:      defaultSnapshot,
+		OrderingField: defaultOrderingField,
+	})
+}
+
+func TestSource_Configure_failure(t *testing.T) {
+	t.Parallel()
+
+	is := is.New(t)
+
+	s := Source{}
+
+	err := s.Configure(context.Background(), map[string]string{
+		config.KeyURI:        "mong\\'odb://localhost:27017",
+		config.KeyDB:         "test",
+		config.KeyCollection: "users",
+	})
+	is.True(err != nil)
+}
+
+func TestSource_Read_success(t *testing.T) {
+	t.Parallel()
+
+	is := is.New(t)
+
+	ctrl := gomock.NewController(t)
+	ctx := context.Background()
+
+	key := make(sdk.StructuredData)
+	key["id"] = 1
+
+	metadata := make(sdk.Metadata)
+	metadata.SetCreatedAt(time.Time{})
+
+	record := sdk.Record{
+		Position: sdk.Position(`{"lastId": 1}`),
+		Metadata: metadata,
+		Key:      key,
+		Payload: sdk.Change{
+			After: key,
+		},
+	}
+
+	it := mock.NewMockIterator(ctrl)
+	it.EXPECT().HasNext(ctx).Return(true, nil)
+	it.EXPECT().Next(ctx).Return(record, nil)
+
+	s := Source{
+		iterator: it,
+	}
+
+	r, err := s.Read(ctx)
+	is.NoErr(err)
+
+	is.Equal(r, record)
+}
+
+func TestSource_Read_failHasNext(t *testing.T) {
+	t.Parallel()
+
+	is := is.New(t)
+
+	ctrl := gomock.NewController(t)
+	ctx := context.Background()
+
+	it := mock.NewMockIterator(ctrl)
+	it.EXPECT().HasNext(ctx).Return(true, errors.New("get data: fail"))
+
+	s := Source{
+		iterator: it,
+	}
+
+	_, err := s.Read(ctx)
+	is.True(err != nil)
+}
+
+func TestSource_Read_failNext(t *testing.T) {
+	t.Parallel()
+
+	is := is.New(t)
+
+	ctrl := gomock.NewController(t)
+	ctx := context.Background()
+
+	it := mock.NewMockIterator(ctrl)
+	it.EXPECT().HasNext(ctx).Return(true, nil)
+	it.EXPECT().Next(ctx).Return(sdk.Record{}, errors.New("key is not exist"))
+
+	s := Source{
+		iterator: it,
+	}
+
+	_, err := s.Read(ctx)
```

Comment:
we can also return the record, and assert that it's nil or empty

Reply:
Updated.